temporalio/contrib/openai_agents/_mcp.py

```diff
@@ -0,0 +1,512 @@
+
+    async def get_prompt(
+        self, name: str, arguments: Optional[dict[str, Any]] = None
+    ) -> GetPromptResult:
+        """Get a specific prompt from the MCP server.
+
+        This method executes a Temporal activity to connect to the MCP server,
+        retrieve the specified prompt with optional arguments, and clean up the connection.
+
+        Args:
+            name: The name of the prompt to retrieve.
+            arguments: Optional dictionary of arguments for the prompt.
+
+        Returns:
+            The prompt result.
+
+        Raises:
+            ActivityError: If the underlying Temporal activity fails.
+        """
+        return await workflow.execute_activity(
+            self.name + "-get-prompt",
+            args=[name, arguments],
+            result_type=GetPromptResult,
+            **self.config,
+        )
+
+    def get_activities(self) -> Sequence[Callable]:
+        """Get the Temporal activities for this MCP server.
+
+        Creates and returns the Temporal activity functions that handle MCP operations.
+        Each activity manages its own connection lifecycle (connect -> operate -> cleanup).
+
+        Returns:
+            A sequence of Temporal activity functions.
+
+        Raises:
+            ValueError: If no MCP server instance was provided during initialization.
+        """
+        server = self.server
+        if server is None:
+            raise ValueError(
+                "A full MCPServer implementation should have been provided when adding a server to the worker."
+            )
+
+        @activity.defn(name=self.name + "-list-tools")
+        async def list_tools() -> list[MCPTool]:
+            try:
+                await server.connect()
+                return await server.list_tools()
+            finally:
+                await server.cleanup()
+
+        @activity.defn(name=self.name + "-call-tool")
+        async def call_tool(
+            tool_name: str, arguments: Optional[dict[str, Any]]
+        ) -> CallToolResult:
+            try:
+                await server.connect()
+                return await server.call_tool(tool_name, arguments)
+            finally:
+                await server.cleanup()
+
+        @activity.defn(name=self.name + "-list-prompts")
+        async def list_prompts() -> ListPromptsResult:
+            try:
+                await server.connect()
+                return await server.list_prompts()
+            finally:
+                await server.cleanup()
+
+        @activity.defn(name=self.name + "-get-prompt")
+        async def get_prompt(
+            name: str, arguments: Optional[dict[str, Any]]
+        ) -> GetPromptResult:
+            try:
+                await server.connect()
+                return await server.get_prompt(name, arguments)
+            finally:
+                await server.cleanup()
+
+        return list_tools, call_tool, list_prompts, get_prompt
+
+
+class StatefulTemporalMCPServer(MCPServer):
```

Comment:
How does this class make use of the parent class? It's a bit surprising that it takes an instance of `MCPServer` in the constructor and also inherits from it.